graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DgsSchemaProvider.kt

```diff
@@ -447,7 +447,7 @@ class DgsSchemaProvider(
                         val instance = env.getObject<Any>()
                         val resolvedType = env.schema.getObjectType(instance::class.java.simpleName)
                         resolvedType
-                            ?: throw InvalidTypeResolverException("The default type resolver could not find a suitable Java type for GraphQL type `${instance::class.java.simpleName}. Provide a @DgsTypeResolver.`")
+                            ?: throw InvalidTypeResolverException("The default type resolver could not find a suitable Java type for GraphQL interface type `$it`. Provide a @DgsTypeResolver for ${instance::class.java.simpleName}.")
```

Comment:
Arguably this will also apply to `UnionTypeDefinition`, so it could be `interface or union type`.
Not sure why are compounding both _interface_ and _union_ concepts here tho.

Reply:
Add union type test and distinguish interface and union